src/Components/CreateImageWizard/steps/Packages/Packages.tsx

```diff
@@ -876,13 +902,35 @@ const Packages = () => {
           );
         }
         setCurrentlyRemovedPackages((prev) =>
-          prev.filter((curr) => curr.name !== pkg.name),
+          prev.filter(
+            (curr) => curr.name !== pkg.name || curr.stream !== pkg.stream,
+          ),
         );
       }
     } else {
-      dispatch(removePackage(pkg.name));
+      dispatch(removePackage(pkg));
       if (pkg.type === 'module' && pkg.module_name) {
         dispatch(removeModule(pkg.module_name));
+        // Check if there will be other selected modules with the same stream AFTER removal
+        const remainingModulesWithSameStream = modules.filter(
+          (m) => m.name !== pkg.module_name && m.stream === pkg.stream,
+        );
+        // Check if there will be other selected packages with the same stream AFTER removal
+        const remainingPackagesWithSameStream = packages.filter(
+          (p) => p.name !== pkg.name && p.stream === pkg.stream,
+        );
+
+        if (
+          pkg.stream === activeStream &&
```

Comment:
I think this is the line @ksiekl referred to here: https://github.com/osbuild/image-builder-frontend/pull/3475#discussion_r2245140787 and I agree. When you unselect package/module the sorting shouldn't reset, but stay as is.

I can't find the mocks anymore, will try to search a bit later. The behavior should be as follow:
- when you select specific stream of a module, all the packages from the active stream should get sorted to the top, followed by all the "other" (packages, other app streams) that should be disabled.
- when you unselect a package, the sorting of the table shouldn't change in any way (so really the only change in the entire table would be that previously checked checkbox will be now unchecked)